libsignal-service/src/sender.rs

```diff
@@ -750,7 +751,25 @@ where
         recipient: Option<&ServiceAddress>,
         data_message: Option<crate::proto::DataMessage>,
         timestamp: u64,
+        send_message_results: &[&SendMessageResult],
     ) -> ContentBody {
+        use sync_message::sent::UnidentifiedDeliveryStatus;
+        let unidentified_status: Vec<UnidentifiedDeliveryStatus> =
+            send_message_results
+                .iter()
+                .filter_map(|result| match result.as_ref() {
+                    Ok(SentMessage {
+                        recipient,
+                        unidentified,
+                        ..
+                    }) => Some(UnidentifiedDeliveryStatus {
+                        destination_e164: recipient.e164(),
+                        destination_uuid: recipient.uuid.map(|s| s.to_string()),
+                        unidentified: Some(*unidentified),
+                    }),
+                    _ => None,
+                })
```

Comment:
I'm thinking that you may be able to get rid of the `match` here:

```suggestion
                .filter_map(Result::ok)
                .map(|sent| {
                    let SentMessage { recipient, unidentified, .. } = sent;
                    UnidentifiedDeliveryStatus {
                        destination_e164: recipient.e164(),
                        destination_uuid: recipient.uuid.map(|s| s.to_string()),
                        unidentified: Some(*unidentified),
                    }
                })
```

Not sure whether it's cleaner or not, and not sure whether it compiles as-is. It says "filter on Ok, then map those to UnidentifiedDeliveryStatus". I think I like it ;p

Reply:
`Result::ok` won't work, but `|result| result.ok()` will because of type inference issues. Doing it now.

Reply:
Not even if you annotate the type?

Reply:
First of all, yes, but why when the compiler can figure it out for you 🔮 I'm not even sure we can do it as nicely, since we're working on a slice, and [need to filter on a reference anyway](https://github.com/whisperfish/libsignal-service-rs/pull/103/commits/26a3665834e26b9340ebf3c5b32089cf7328d2c1#diff-9ae521e6fffe43a399d2d3007bbccc81a73e245f5aa5663b007af1fa97dbcc26L760).

Reply:
I don't see why it needs to be a reference, but if you say it has to be, then it's alright with me now.